### PR TITLE
feat: Unordered ParallelSelect

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
Adds an `unordered` options to ParallelSelect.

### CODE REVIEW

File: csharp/src/ParallelSelectInternal.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    }
+    finally
+    {
+      // Ensure all running tasks are actually aborted
+      globalCts.Cancel();
+    }
+
+    sem.Dispose();
+    iterationCts.Dispose();
+    globalCts.Dispose();
```
Comment (aneojgurhem) [inline csharp/src/ParallelSelectInternal.cs:126]: Why aren't they in the finally ? Or declared with `using var` ?

**Review by aneojgurhem — APPROVED** [review]
Very nice implementation using channels. It simplifies the code a lot

File: csharp/src/ParallelSelectInternal.cs (hunk 1 above)
Comment (lemaitre-aneo) [reply csharp/src/ParallelSelectInternal.cs:126]: If there is an error, there are some tasks in-flight that still need those.

Comment (aneojgurhem) [reply csharp/src/ParallelSelectInternal.cs:126]: It should not change this behavior in the finally just above while ensuring that it is called

Comment (lemaitre-aneo) [reply csharp/src/ParallelSelectInternal.cs]: Indeed, it does work. Changed.

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**